src/order/cover.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Violeta Hernández Palacios, Grayson Burton
+-/
+import data.set.intervals.basic
+
+/-!
+# The covering relation
+
+This file defines the covering relation in an order. `b` is said to cover `a` if `a < b` and there
+is no element in between.
+
+## Notation
+
+`a ⋖ b` means that `b` covers `a`.
+-/
+
+open set
+
+variables {α : Type*}
+
+section has_lt
+variables [has_lt α] {a b : α}
+
+/-- `covers a b` means that `b` covers `a`: `a < b` and there is no element in between. -/
+def covers (a b : α) : Prop := a < b ∧ ∀ ⦃c⦄, a < c → ¬ c < b
+
+infix ` ⋖ `:50 := covers
+
+lemma covers.lt (h : a ⋖ b) : a < b := h.1
+
+/-- If `a < b`, then `b` does not cover `a` iff there's an element in between. -/
+lemma not_cover_iff (h : a < b) : ¬a ⋖ b ↔ ∃ c, a < c ∧ c < b :=
+by { simp_rw [covers, not_and, not_forall, exists_prop, not_not], exact imp_iff_right h }
+
+open order_dual
+
+@[simp] lemma to_dual_cover_to_dual_iff : to_dual b ⋖ to_dual a ↔ a ⋖ b :=
+and_congr_right' $ forall_congr $ λ c, forall_swap
```

Comment:
Some lemmas use `cover`, some use `covers`. Also, I think that we should have `covers.to_dual` (e.g., using `alias to_dual_cover_to_dual_iff ↔ _ covers.to_dual`).